### PR TITLE
Word wrappers

### PR DESCRIPTION
**Pull request by anne-gropler**
A label can have a maximum line width (`label.lineWidth`). If the label exceeds this line width, the word wrapper mode decides whether the label goes on a next line (`newLine`) or wether the label gets abbreviated, using an `Ellipsis` (e.g., "This is too long" --> "This is t..")

The `wordWrap` property is removed from label, since its functionality is already covered by the `wordWrapper` property. (`wordWrap=false` is now `wordWrapper.None`). This removes ambiguity. This is a _breaking change_.

### CODE REVIEW

**Comment by anne-gropler** [comment]
Known issue:
Line feeds are not handled well when they appear in the last line (of a multi-line label) and when that line fits the label.lineWidth, although the lines before would not fit. This is documented in code. I don't think it is urgent to fix, since the use case for that might not happen that much.

File: source/text/typesetter.ts [hunk 1]
```diff
@@ -323,7 +323,7 @@ export class Typesetter {
                 /** Update the label's text. We cannot undo this.
                  * @todo make it undoable? e.g., label.originalText and label.currentText ?
                  */
-                newText = newText.slice(0, index) + label.ellpsisChars + newText.slice(index + 1);
+                newText = newText.slice(0, index + 1) + label.ellpsisChars + newText.slice(index + 1);
```
Comment (scheibel) [inline source/text/typesetter.ts]: Seems like there is a typo in `ellpsisChars`. As of now, I cant say anything about the algorithm.

File: source/text/label.ts [hunk 2]
```diff
@@ -17,10 +17,13 @@ import { Text } from './text';
...
     /** @see {@link text} */
-    protected _text: Text | string;
+    protected _text: Text;
 
-    /** @see {@link wordWrap} */
-    protected _wordWrap = false;
+    /** @see {@link wordWrapper} */
+    protected _wordWrapper: Label.WordWrapper = Label.WordWrapper.None;
+
+    /** @see {@link ellpsisChars} */
```
Comment (scheibel) [inline source/text/label.ts]: Type here, too.

Comment (anne-gropler) [reply source/text/label.ts]: haha, thank you. Yes, this is a copy-paste-typo :D

**Pull request merged**